assets/javascripts/overview.js

```diff
@@ -287,8 +287,34 @@ function showAddCommentsDialog() {
   modal.show();
 }
 
-function addComments(form) {
+function restartJobs(btn) {
+  const form = btn.closest('form');
   const text = form.elements.text.value;
+  const jobs = btn.dataset.jobs
+    .split(',')
+    .map(Number)
+    .filter(n => !isNaN(n));
+  const apiUrl = btn.dataset.url;
+  if (!text.length) {
+    return window.alert("The comment text mustn't be empty.");
+  }
+
+  const progressIndication = document.getElementById('add-comments-progress-indication');
+  const controls = document.getElementById('add-comments-controls');
+  progressIndication.style.display = 'flex';
+  controls.style.display = 'none';
+  restartJob(apiUrl, jobs, text);
+  setTimeout(() => {
+    progressIndication.style.display = 'none';
+    controls.style.display = 'inline';
+    window.addCommentsModal.hide();
```

Comment:
It would actually be nice to have some kind of progress indication because restarting many jobs and commenting on them might take a while. Maybe we can keep the modal open (and only hiding the buttons) and use the existing progress indication (that is part of the modal when I remember correctly). This of course means you'll need to close the modal when `restartJob` is done by making sure `restartJob` returns the promise and using the `finally` handler of it.

Reply:
I tried to add a progress bar. but it turns to not be as simple and further changes are required. for instance to the `restartJob` IMO (as you mentioned to return the promise, but I dont think is enough). Needs some thought.
Anyway, I removed the `setTimeout` to simplify the flow and address your concerns here and on  https://github.com/os-autoinst/openQA/pull/5971#discussion_r1816650937